internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr_test.go

```diff
@@ -1469,3 +1527,609 @@ func TestKafkaManager_calculateAvailableAndMadCapacityForEnterpriseClusters(t *t
+							*updatedExpiresAt = values["expires_at"].(sql.NullTime)
+							return nil
+						},
+					}
+				},
+			},
+			args: args{
+				kafkas: dbapi.KafkaList{
+					{
+						Meta: api.Meta{
+							CreatedAt: time.Now(),
+						},
+						InstanceType:            "instance-type-1",
+						SizeId:                  "x1",
+						Status:                  constants.KafkaRequestStatusReady.String(),
+						ActualKafkaBillingModel: "kafka-bm1",
+						ExpiresAt: sql.NullTime{
+							Time:  time.Time{},
+							Valid: true,
+						},
+					},
+				},
+			},
+			wantNewExpiresAtValue: sql.NullTime{Time: time.Now().Add(time.Duration(172800) * time.Second), Valid: true},
+			wantUpdateCallCount:   1,
+			wantErrCount:          0,
+		},
+		{
+			name: "should skip updated based on lifespanSeconds if defined if expires_at is already set",
+			fields: fields{
+				kafkaConfig: &config.KafkaConfig{
+					SupportedInstanceTypes: &config.KafkaSupportedInstanceTypesConfig{
+						Configuration: config.SupportedKafkaInstanceTypesConfig{
+							SupportedKafkaInstanceTypes: []config.KafkaInstanceType{
+								{
+									Id: "instance-type-1",
+									Sizes: []config.KafkaInstanceSize{
+										{
+											Id:              "x1",
+											LifespanSeconds: &[]int{172800}[0],
+										},
+									},
+									SupportedBillingModels: []config.KafkaBillingModel{
+										{
+											ID: "kafka-bm1",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				kafkaService: func(updatedExpiresAt *sql.NullTime, updateCallCount *int) services.KafkaService {
+					return &services.KafkaServiceMock{
+						UpdatesFunc: func(kafkaRequest *dbapi.KafkaRequest, values map[string]interface{}) *errors.ServiceError {
+							*updateCallCount += 1
+							*updatedExpiresAt = values["expires_at"].(sql.NullTime)
+							return nil
+						},
+					}
+				},
+			},
+			args: args{
+				kafkas: dbapi.KafkaList{
+					{
+						InstanceType:            "instance-type-1",
+						SizeId:                  "x1",
+						Status:                  constants.KafkaRequestStatusReady.String(),
+						ActualKafkaBillingModel: "kafka-bm1",
+						ExpiresAt: sql.NullTime{
+							Time:  time.Now(),
+							Valid: true,
+						},
+					},
+				},
+			},
+			wantNewExpiresAtValue: sql.NullTime{},
+			wantUpdateCallCount:   0,
+			wantErrCount:          0,
+		},
+	}
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+			t.Parallel()
+			updatedExpiresAt := sql.NullTime{}
+			updateCallCount := 0
+			k := &KafkaManager{
+				kafkaService: tt.fields.kafkaService(&updatedExpiresAt, &updateCallCount),
+				kafkaConfig:  tt.fields.kafkaConfig,
+			}
+			err := k.reconcileKafkaExpiresAt(tt.args.kafkas)
+			g.Expect(len(err)).To(gomega.Equal(tt.wantErrCount))
+			g.Expect(updateCallCount).To(gomega.Equal(tt.wantUpdateCallCount), "expected update call count does not match actual")
```

Comment:
I think the Mock type provides a way to check how many times the Updates call is called. Can you review that?
I think we can avoid using that counter

Reply:
you're right, i completely missed that. Thanks! I'll update this to use that instead